cfavml/src/safe_trait_numeric_ops.rs

```diff
@@ -0,0 +1,85 @@
+//! Safe but somewhat low-level variants of the arithmetic operations in CFAVML.
+//!
+//! In general, I would recommend using the higher level generic functions api which provides
+//! some syntax sugar over these traits.
+
+use crate::buffer::WriteOnlyBuffer;
+use crate::danger::export_hypot;
+use crate::mem_loader::{IntoMemLoader, MemLoader};
+
+/// Various arithmetic operations over vectors.
+pub trait NumericOps: Sized + Copy {
```

Comment:
A little bit of bike shedding; If this is called `NumericOps`, does this open us up to a bit of pain if later we want to add a function that falls under this category, but for `i32, i8, etc...` Since here `NumericOps` is probably a bit more like `FloatOps` or similar.

Reply:
No worries on the bike shedding. 

What functions would fall into this category that would be valid for integers? I can change the name of the trait/file to hypot, but I imagine there would be other float specific ops that would be useful to add at some point.

What naming scheme do you think we should go with?

Reply:
Not sure, I suspect that, we are more likely to have Float-only ops rather than int-only ops most of the time for this lib, A trait for just the hypot safe op is probably going to become a bit verbose pretty quickly, we could put it under `MiscFloatOps` as a bit of a catchall and then come up with a better fitting name if we collect a few more functions under this category/applicaiton.